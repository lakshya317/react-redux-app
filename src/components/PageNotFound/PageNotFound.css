.page-body{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    flex: 1 1 auto;
}

.page-title{
    display: flex;
    font-size: 60px;
    font-weight: 1000;
    margin-bottom: 40px;
}

.page-title-sm{
    display: flex;
    font-size: 30px;
    font-weight: 800;
    margin-bottom: 40px;
}

.loading-not-found{
    display: flex;
}