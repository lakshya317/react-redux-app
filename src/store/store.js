import { createStore } from 'redux';
import { persistStore, persistReducer } from 'redux-persist';
import { encryptTransform } from 'redux-persist-transform-encrypt';
import storage from 'redux-persist/lib/storage';
import reducer from '../reducer';
import middleware from '../middleware';

// Config
const persistStoreEncryptionKey = process.env.REACT_APP_PERSIST_STORE_ENCRYPTION_KEY;
const persistConfigSecret = process.env.REACT_APP_PERSIST_CONFIG_SECRET;

//Encryptor
const encryptor = encryptTransform({
    secretKey: persistStoreEncryptionKey,
    onError: function (error) {
        // Handle the error.
        console.error(error);
    }
});

const persistConfig = {
    transforms: [encryptor],
    storage,
    key: persistConfigSecret
};

const persistedReducer = persistReducer(persistConfig, reducer);

export const store = createStore(persistedReducer, middleware);
export const persistor = persistStore(store);
