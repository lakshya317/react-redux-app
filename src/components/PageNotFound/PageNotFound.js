import React, {useEffect} from 'react';
import { useNavigate } from 'react-router-dom';
import "./PageNotFound.css";
import Header from "../Header/Header";

const PageNotFound = () => {
    const navigate = useNavigate();

    useEffect(()=>{
        setTimeout(()=>{
            navigate("/react-redux-app/")
        }, 2000)
    },[navigate])

    return (
        <div className="landing">
            <Header />
            <div className='page-body'>
                <div className='page-title'>
                    Error 404: Page Not Found.
                </div>
                <div className='page-title-sm'>
                    Redirecting you to Home Page...
                </div>
            </div>
        </div>
    );
}

export default PageNotFound;
