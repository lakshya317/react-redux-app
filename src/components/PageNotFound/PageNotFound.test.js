import { render } from '@testing-library/react';
import PageNotFound from './PageNotFound';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from "../../store/store";

describe('PageNotFound Tests', () => {

    test('PageNotFound Snapshot Test', () => {
        const notFound = render(
            <Provider store={store}>
                <MemoryRouter>
                    <PageNotFound current={2} perPage={6} total={17} totalPages={3} handlePageChange={(newPage)=>{console.log(newPage)}}/>
                </MemoryRouter>
            </Provider>
        );

        expect(notFound).toMatchSnapshot()
    });
});