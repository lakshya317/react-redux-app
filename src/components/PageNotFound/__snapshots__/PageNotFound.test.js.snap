// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound Tests PageNotFound Snapshot Test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="landing"
      >
        <div
          class="header"
        >
          <a
            href="/"
            style="text-decoration: none;"
          >
            <img
              alt="Shell Logo"
              class="Shell-logo"
              src="Shell_Logo.png"
            />
          </a>
        </div>
        <div
          class="page-body"
        >
          <div
            class="page-title"
          >
            Error 404: Page Not Found!
          </div>
          <div
            class="page-title-sm"
          >
            Redirecting you to Home Page...
          </div>
          <div
            class="loading-not-found"
          >
            <div
              class="spinner-container"
            >
              <div
                class="spinner-border text-dark"
                role="status"
              >
                <span
                  class="visually-hidden"
                >
                  Loading...
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="landing"
    >
      <div
        class="header"
      >
        <a
          href="/"
          style="text-decoration: none;"
        >
          <img
            alt="Shell Logo"
            class="Shell-logo"
            src="Shell_Logo.png"
          />
        </a>
      </div>
      <div
        class="page-body"
      >
        <div
          class="page-title"
        >
          Error 404: Page Not Found!
        </div>
        <div
          class="page-title-sm"
        >
          Redirecting you to Home Page...
        </div>
        <div
          class="loading-not-found"
        >
          <div
            class="spinner-container"
          >
            <div
              class="spinner-border text-dark"
              role="status"
            >
              <span
                class="visually-hidden"
              >
                Loading...
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
